### PR TITLE
Update Jujulib Identity API to use the user instance to fetch identity URL

### PR DESCRIPTION
**Pull request by hatched**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8405/

File: jujugui/static/gui/src/app/jujulib/identity.js [hunk 1]
```diff
@@ -19,28 +19,40 @@ var module = module;
...
+  function identity(userStorage, bakery) {
     // Store the API URL (including version) handling missing trailing slash.
-    this.url = url.replace(/\/?$/, '/') + identityAPIVersion;
+    this.userStorage = userStorage;
     this.bakery = bakery;
   };
 
   identity.prototype = {
 
+    getIdentityURL: function() {
```
Comment (frankban) [inline jujugui/static/gui/src/app/jujulib/identity.js]: Docstring?

File: jujugui/static/gui/src/app/jujulib/identity.js [hunk 2]
```diff
@@ -19,28 +19,40 @@ var module = module;
     Initializer.
 
     @function identity
-    @param url {String} The URL of the identity instance, including
-      scheme and port, and excluding the API version.
+    @param userStorage {Object} An object containing access to the identity url.
```
Comment (frankban) [inline jujugui/static/gui/src/app/jujulib/identity.js]: /s/url/URL/ [shrug]

File: jujugui/static/gui/src/app/jujulib/identity.js [hunk 3]
```diff
@@ -19,28 +19,40 @@ var module = module;
...
       Fetch the user data from the identity service
       @param {String} userName The user name to fetch the data for.
       @param {Function} callback
         Called with the call signature (error <Object>, userData <Object>).
     */
     getUser: function(userName, callback) {
-      return this.bakery.get(`${this.url}/u/${userName}`, null, (err, resp) => {
+      const identityURL = this.getIdentityURL();
+      if (identityURL === null) {
+        callback('no identity url available', null);
```
Comment (frankban) [inline jujugui/static/gui/src/app/jujulib/identity.js]: s/url/URL/ this is likely to be user facing.

File: jujugui/static/gui/src/app/jujulib/identity.js (hunk 2 above)
Comment (hatched) [reply jujugui/static/gui/src/app/jujulib/identity.js]: done

File: jujugui/static/gui/src/app/jujulib/identity.js (hunk 1 above)
Comment (hatched) [reply jujugui/static/gui/src/app/jujulib/identity.js]: oops, rushed, fixed

File: jujugui/static/gui/src/app/jujulib/identity.js (hunk 3 above)
Comment (hatched) [reply jujugui/static/gui/src/app/jujulib/identity.js]: done

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8406/

**Pull request merged**